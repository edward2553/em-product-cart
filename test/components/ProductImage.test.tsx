import renderer from 'react-test-renderer';
import React from 'react';
import ProductCart, { ProductImage } from '../../src/components';
import { product2 } from '../data.ts/products';

describe('ProductImage to have the url passed and match the snapshot', () => {
    test('It should have the url passed by props', () => {
        const wrapper = renderer.create(<ProductImage img={product2.img} />)
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
    test('Debe de mostrar el componente correctamente con el nombre del producto', () => {
        const wrapper = renderer.create(
          <ProductCart product={product2}>{() => <ProductImage />}</ProductCart>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
      });
});