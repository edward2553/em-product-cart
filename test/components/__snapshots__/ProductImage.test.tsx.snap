// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage to have the url passed and match the snapshot Debe de mostrar el componente correctamente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage to have the url passed and match the snapshot It should have the url passed by props 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="./coffee-mug.png"
/>
`;
